packages/medusa/src/loaders/models.ts

```diff
@@ -5,30 +5,76 @@ import { ClassConstructor, MedusaContainer } from "../types/global"
 import { EntitySchema } from "typeorm"
 import { asClass, asValue } from "awilix"
 
+function formatModelName(modelName) {
+  // Remove "Model" from the end of the string
+  let formattedName = modelName.replace('Model', '');
+
+  // Capitalize the first letter of the string
+  formattedName = formattedName.charAt(0).toUpperCase() + formattedName.slice(1);
+
+  return formattedName;
+}
+
 /**
  * Registers all models in the model directory
  */
 export default (
-  { container, isTest }: { container: MedusaContainer; isTest?: boolean },
+  { container, isTest, rootDirectory }: { container: MedusaContainer; isTest?: boolean, rootDirectory?: string },
   config = { register: true }
 ) => {
   const corePath = isTest ? "../models/*.ts" : "../models/*.js"
+  const modelExtensionsGlob = "dist/models/*.js"
```

Comment:
todo: should we use the same glob as we do in the plugins loader

Reply:
The plugin models are handled separately in a different loader, it would make sense to keep the plugin model extending in that context.

Reply:
Both plugins and backend models are loaded from the plugins loaders

Reply:
as discussed, added to both places.